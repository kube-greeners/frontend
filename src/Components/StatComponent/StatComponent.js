import React from "react";
import { Card } from "antd";

function StatComponent({
  gridArea,
  title,
  success1,
  success2,
  stat1,
  stat2,
  unit,
  integer = false,
}) {
  const statContainerStyle = {
    flex: "1",
    textAlign: "center",
    fontWeight: "bold",
  };
  
 const decimals = integer? 0 : 2;

  return (
    <>
      <Card style={{ ...statContainerStyle, gridArea: gridArea }} title={title}>
        {success2
          ? success1 && success2
            ? `${stat1.toFixed(decimals)} ${unit} / ${stat2.toFixed(decimals)} ${unit}`
            : "Loading..."
          : success1
          ? `${stat1.toFixed(decimals)} ${unit ? unit : ""}`
          : "Loading..."}
      </Card>
    </>
  );
}

export default StatComponent;
