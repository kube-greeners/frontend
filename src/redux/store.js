import { configureStore } from '@reduxjs/toolkit'
import dashboardReducer from './dashboardSlice'
import { apiSlice } from './apiSlice'

export default configureStore({
  reducer: {
    dashboard: dashboardReducer,
    [apiSlice.reducerPath]:apiSlice.reducer
  },
  middleware: getDefaultMiddleware =>
  getDefaultMiddleware().concat(apiSlice.middleware)
})