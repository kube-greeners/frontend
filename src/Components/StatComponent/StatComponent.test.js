import StatComponent from './StatComponent';
import renderer from 'react-test-renderer'
import store from "../../redux/store";
import { Provider } from "react-redux";

test("matches snapshot", () => {
  const tree = renderer.create(<Provider store={store}><StatComponent/></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
