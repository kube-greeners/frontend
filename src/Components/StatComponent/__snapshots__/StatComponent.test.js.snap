// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches StatComponent snapshot 1`] = `
<div
  className="ant-card ant-card-bordered"
  style={
    Object {
      "flex": "1",
      "fontFamily": "Helvetica",
      "fontSize": "17px",
      "fontWeight": "bold",
      "gridArea": undefined,
      "textAlign": "center",
      "textShadow": "1px 0px 1px",
    }
  }
>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    Loading...
  </div>
</div>
`;
